modules/javafx.graphics/src/test/java/test/javafx/scene/Node_transition_Test.java

```diff
@@ -0,0 +1,252 @@
+    }
+
+    @Test
+    public void testPropertyNameIsCaseSensitive() {
+        CssMetaData<?, ?> opacityProperty = Node.getClassCssMetaData().stream()
+            .filter(md -> md.getProperty().equals("-fx-opacity"))
+            .findFirst()
+            .orElseThrow();
+
+        node.setStyle("transition: -fx-OPACITY 1s");
+        node.applyCss();
+        assertNull(NodeHelper.findTransitionDefinition(node, opacityProperty));
+
+        node.setStyle("transition: -fx-opacity 1s");
+        node.applyCss();
+        assertNotNull(NodeHelper.findTransitionDefinition(node, opacityProperty));
+    }
+
+    @Test
+    public void testAllIdentifierIsCaseInsensitive() {
+        node.setStyle("transition: ALL 1s");
+        node.applyCss();
+        List<TransitionDefinition> transitions = NodeShim.getTransitionDefinitions(node);
+        assertEquals(1, transitions.size());
+        assertTransitionEquals("all", Duration.seconds(1), Duration.ZERO, CSS_EASE, transitions.get(0));
+
+        node.setStyle("transition: all 1s");
+        node.applyCss();
+        transitions = NodeShim.getTransitionDefinitions(node);
+        assertEquals(1, transitions.size());
+        assertTransitionEquals("all", Duration.seconds(1), Duration.ZERO, CSS_EASE, transitions.get(0));
+    }
+
+    @Test
+    public void testLastOccurrenceOfMultiplyReferencedPropertyIsSelected() {
+        node.setStyle("transition: -fx-fill 1s, -fx-fill 2s ease-in-out;");
+        node.applyCss();
+
+        CssMetaData<?, ?> propertyMetadata = node.getCssMetaData().stream()
+            .filter(m -> m.getProperty().equals("-fx-fill"))
+            .findFirst()
+            .orElseThrow();
+        TransitionDefinition transition = NodeHelper.findTransitionDefinition(node, propertyMetadata);
+        assertTransitionEquals("-fx-fill", Duration.seconds(2), Duration.ZERO, CSS_EASE_IN_OUT, transition);
+    }
+
+    @Test
+    public void testTransitionsAreAppliedWhenPseudoClassIsChanged() {
+        String url = "data:text/css;base64," + Base64.getUrlEncoder().encodeToString("""
+            .testClass {
+                -fx-background-color: green;
+                transition: -fx-background-color 1s 0.5s ease, -fx-scale-x 1s, -fx-scale-y 1s;
+            }
+
+            .testClass:hover {
+                -fx-background-color: red;
+                -fx-scale-x: 1.2;
+                -fx-scale-y: 1.2;
+                transition: -fx-background-color 1s;
+            }
+            """.getBytes(StandardCharsets.UTF_8));
+
+        scene.getStylesheets().add(url);
+        node.getStyleClass().add("testClass");
+        node.applyCss();
+
+        List<TransitionDefinition> transitions = NodeShim.getTransitionDefinitions(node);
+        assertEquals(3, transitions.size());
+        assertTransitionEquals("-fx-background-color", Duration.seconds(1), Duration.seconds(0.5), CSS_EASE, transitions.get(0));
```

Comment:
`node` is a `Rectangle`, which doesn't have a `background`property. Doesn't hurt this specific test, but maybe it's better to use an existing property.

Reply:
I've changed the test to use `-fx-fill` instead.